src/vivarium/framework/population.py

```diff
@@ -220,15 +222,22 @@ def _order_initializers(self) -> None:
             starting_length = len(unordered_initializers)
             for _ in range(len(unordered_initializers)):
                 initializer, columns_created, columns_required = unordered_initializers.pop()
+                initializer_created.extend(columns_created)
+                initializer_required.extend(columns_required)
                 if set(columns_required) <= set(available_columns):
                     self._population_initializers.append((initializer, columns_created, columns_required))
                     available_columns.extend(columns_created)
                 else:
                     unordered_initializers.appendleft((initializer, columns_created, columns_required))
 
         if unordered_initializers:
-            raise PopulationError(f"The initializers {unordered_initializers} could not be added.  "
-                                  f"Check for cyclic dependencies in your components.")
+
```

Comment:
You typically want to create variables as close as possible to where you're using them.  Try:

```
if unordered_initializers:
    _, initializer_required, initializer_created = zip(*unordered_initializers)
    if set(initializer_required)...
```

Reply:
when comparing requires and creates, can I safely ignore tracked?